tests/unit/test_checkpointing.py

```diff
@@ -1190,13 +1190,22 @@ def _go():
             loss = model(batch[0], batch[1])
             model.backward(loss)
             model.step()
+        if load_optim:
+            torch.save(model.optimizer.optimizer.state_dict(),
+                       os.path.join(tmpdir,
+                                    'opt-state-dict'))
         model.save_checkpoint(tmpdir)
 
         ds_config["zero_optimization"]["elastic_checkpoint"] = elastic_load
         model, _, _, _ = deepspeed.initialize(config=ds_config,
                                               model=models[1],
                                               model_parameters=models[1].parameters())
         model.load_checkpoint(tmpdir, load_optimizer_states=load_optim)
+
+        if load_optim:
+            saved_sd = torch.load(os.path.join(tmpdir, 'opt-state-dict'))
+            curr_sd = model.optimizer.optimizer.state_dict()
+            assert curr_sd['param_groups'] == saved_sd['param_groups']
```

Comment:
so one workflow fails here: https://github.com/microsoft/DeepSpeed/runs/7184872087?check_suite_focus=true

mismatching on the number of params:

```
   -   'params': [139739444197968],
  ?                     ^^ ^^^ -
  +   'params': [139739450210816],
  ?                     ^^^ ^^^
```